--- imlib-1.9.14/configure.in.orig	Wed May  1 21:54:31 2002
+++ imlib-1.9.14/configure.in	Wed May  1 21:59:55 2002
@@ -233,21 +233,21 @@
 
 dnl Test for libtiff
 AC_CHECK_LIB(tiff, TIFFReadScanline,
-  AC_CHECK_HEADER(tiffio.h,
+  [AC_CHECK_HEADER(tiffio.h,[
     TIFFLIBS="-ltiff"
-    SUPPORT_LIBS="$SUPPORT_LIBS -ltiff"; AC_DEFINE(HAVE_LIBTIFF),
-    AC_MSG_ERROR(*** TIFF header files not found ***)),
-  AC_CHECK_LIB(tiff, TIFFWriteScanline,
-    AC_CHECK_HEADER(tiffio.h,
+    SUPPORT_LIBS="$SUPPORT_LIBS -ltiff"; AC_DEFINE(HAVE_LIBTIFF)],
+    AC_MSG_ERROR([*** TIFF header files not found ***]))],
+  [AC_CHECK_LIB(tiff, TIFFWriteScanline,
+    [AC_CHECK_HEADER(tiffio.h,[
       TIFFLIBS="-ltiff -ljpeg -lz"
-      SUPPORT_LIBS="$SUPPORT_LIBS -ltiff -ljpeg -lz"; AC_DEFINE(HAVE_LIBTIFF),
-      AC_MSG_ERROR(*** TIFF header files not found ***)),
-    AC_CHECK_LIB(tiff34, TIFFFlushData,
-      AC_CHECK_HEADER(tiffio.h,
+      SUPPORT_LIBS="$SUPPORT_LIBS -ltiff -ljpeg -lz"; AC_DEFINE(HAVE_LIBTIFF)],
+      AC_MSG_ERROR([*** TIFF header files not found ***]))],
+    [AC_CHECK_LIB(tiff34, TIFFFlushData,
+      [AC_CHECK_HEADER(tiffio.h,[
         TIFFLIBS="-ltiff34 -ljpeg -lz"
-        SUPPORT_LIBS="$SUPPORT_LIBS -ltiff34 -ljpeg -lz"; AC_DEFINE(HAVE_LIBTIFF),
-        AC_MSG_ERROR(*** TIFF header files not found ***)),
-    AC_MSG_ERROR(*** TIFF library not found ***), -ljpeg -lz $GX_LIBS), -ljpeg -lz $GX_LIBS), $GX_LIBS)
+        SUPPORT_LIBS="$SUPPORT_LIBS -ltiff34 -ljpeg -lz"; AC_DEFINE(HAVE_LIBTIFF)],
+        AC_MSG_ERROR([*** TIFF header files not found ***]))],
+    AC_MSG_ERROR([*** TIFF library not found ***]), [-ljpeg -lz $GX_LIBS])], [-ljpeg -lz $GX_LIBS])], [$GX_LIBS])
 
 
 AC_CHECK_HEADER(gif_lib.h,
@@ -265,10 +265,10 @@
   AC_MSG_ERROR(*** GIF header not found ***))
 
 AC_CHECK_LIB(png, png_read_info,
-  AC_CHECK_HEADER(png.h,
+  [AC_CHECK_HEADER(png.h,
     png_ok=yes,
-    png_ok=no),
-  AC_MSG_ERROR(*** PNG library not found ***), -lz -lm $GX_LIBS)
+    png_ok=no)],
+  AC_MSG_ERROR([*** PNG library not found ***]), [-lz -lm $GX_LIBS])
 if test "$png_ok" = yes; then
   AC_MSG_CHECKING([for png_structp in png.h])
   AC_TRY_COMPILE([#include <png.h>],
