--- imlib-1.9.14/configure.in.orig	Wed May  1 19:15:02 2002
+++ imlib-1.9.14/configure.in	Wed May  1 23:47:14 2002
@@ -233,21 +233,21 @@
 
 dnl Test for libtiff
 AC_CHECK_LIB(tiff, TIFFReadScanline,
-  AC_CHECK_HEADER(tiffio.h,
-    TIFFLIBS="-ltiff"
-    SUPPORT_LIBS="$SUPPORT_LIBS -ltiff"; AC_DEFINE(HAVE_LIBTIFF),
-    AC_MSG_ERROR(*** TIFF header files not found ***)),
-  AC_CHECK_LIB(tiff, TIFFWriteScanline,
-    AC_CHECK_HEADER(tiffio.h,
-      TIFFLIBS="-ltiff -ljpeg -lz"
-      SUPPORT_LIBS="$SUPPORT_LIBS -ltiff -ljpeg -lz"; AC_DEFINE(HAVE_LIBTIFF),
-      AC_MSG_ERROR(*** TIFF header files not found ***)),
-    AC_CHECK_LIB(tiff34, TIFFFlushData,
-      AC_CHECK_HEADER(tiffio.h,
-        TIFFLIBS="-ltiff34 -ljpeg -lz"
-        SUPPORT_LIBS="$SUPPORT_LIBS -ltiff34 -ljpeg -lz"; AC_DEFINE(HAVE_LIBTIFF),
-        AC_MSG_ERROR(*** TIFF header files not found ***)),
-    AC_MSG_ERROR(*** TIFF library not found ***), -ljpeg -lz $GX_LIBS), -ljpeg -lz $GX_LIBS), $GX_LIBS)
+  [AC_CHECK_HEADER(tiffio.h,
+    [TIFFLIBS="-ltiff"
+    SUPPORT_LIBS="$SUPPORT_LIBS -ltiff"; AC_DEFINE(HAVE_LIBTIFF)],
+    AC_MSG_ERROR([*** TIFF header files not found ***]))],
+  [AC_CHECK_LIB(tiff, TIFFWriteScanline,
+    [AC_CHECK_HEADER(tiffio.h,
+      [TIFFLIBS="-ltiff -ljpeg -lz"
+      SUPPORT_LIBS="$SUPPORT_LIBS -ltiff -ljpeg -lz"; AC_DEFINE(HAVE_LIBTIFF)],
+      AC_MSG_ERROR([*** TIFF header files not found ***]))],
+    [AC_CHECK_LIB(tiff34, TIFFFlushData,
+      [AC_CHECK_HEADER(tiffio.h,
+        [TIFFLIBS="-ltiff34 -ljpeg -lz"
+        SUPPORT_LIBS="$SUPPORT_LIBS -ltiff34 -ljpeg -lz"; AC_DEFINE(HAVE_LIBTIFF)],
+        AC_MSG_ERROR([*** TIFF header files not found ***]))],
+    AC_MSG_ERROR([*** TIFF library not found ***]), p-ljpeg -lz $GX_LIBS]), [-ljpeg -lz $GX_LIBS])], [$GX_LIBS])
 
 
 AC_CHECK_HEADER(gif_lib.h,
