--- imlib-1.9.15/Imlib/Makefile.am.orig	2004-09-21 02:22:59.000000000 +0200
+++ imlib-1.9.15/Imlib/Makefile.am	2006-08-27 00:05:56.084387000 +0200
@@ -32,7 +32,7 @@
 
 libImlib_la_LDFLAGS    = -version-info 10:15:9
 #libImlib_la_LDDADD     = $(SUPPORT_LIBS)
-libImlib_la_LIBADD     = $(SUPPORT_LIBS) $(X_PRE_LIBS) \
+libImlib_la_LIBADD     = $(SUPPORT_LIBS) \
 			 $(X_LIBS) $(X_EXTRA_LIBS)
 
 DEFS =  -DSYSTEM_IMRC=\"$(sysconfdir)/imrc\" \
--- imlib-1.9.15/configure.in.orig	2006-08-27 00:03:59.173080000 +0200
+++ imlib-1.9.15/configure.in	2006-08-27 00:05:30.226771000 +0200
@@ -137,7 +137,7 @@
 if test -z "$GDK_IMLIB"; then
   # see macros/gnome-x-checks.m4, and gtk.m4
   CPPFLAGS="$CPPFLAGS $X_CFLAGS"
-  GX_LIBS="$X_PRE_LIBS $X_LIBS -lX11 -lm"
+  GX_LIBS="$X_LIBS -lX11 -lm"
 else
   CPPFLAGS="$CPPFLAGS $GTK_CFLAGS"
   GX_LIBS="$GTK_LIBS"
--- imlib-1.9.15/imlib-config.in.orig	2006-08-27 01:18:27.940361000 +0200
+++ imlib-1.9.15/imlib-config.in	2006-08-27 01:20:09.714721500 +0200
@@ -59,7 +59,7 @@
       ;;
     --libs)
       libdirs=-L@libdir@
-      echo $libdirs -lImlib $gx_libs @X_LIBS@ @X_PRE_LIBS@ -lXext -lX11 @X_EXTRA_LIBS@
+      echo $libdirs -lImlib $gx_libs @X_LIBS@ -lXext -lX11 @X_EXTRA_LIBS@
       ;;
     --libs-gdk)
       libdirs=-L@libdir@
--- imlib-1.9.15/imlib.pc.in.orig	2004-08-27 18:03:11.000000000 +0200
+++ imlib-1.9.15/imlib.pc.in	2006-08-27 01:20:29.887982250 +0200
@@ -9,5 +9,5 @@
 Description: An image loading and rendering library for X11R6
 Version: @VERSION@
 Requires:
-Libs: -L${libdir} @LDFLAGS@ -lImlib @SUPPORT_LIBS@ @X_LIBS@ @X_PRE_LIBS@ -lXext -lX11 @X_EXTRA_LIBS@
+Libs: -L${libdir} @LDFLAGS@ -lImlib @SUPPORT_LIBS@ @X_LIBS@ -lXext -lX11 @X_EXTRA_LIBS@
 Cflags: -I${includedir} @X_CFLAGS@
